src/mappings/nfts/buy.ts

```diff
@@ -15,18 +15,20 @@ export async function handleTokenBuy(context: Context): Promise<void> {
   debug(OPERATION, event, true)
 
   const id = createTokenId(event.collectionId, event.sn)
-  const entity = await getWith(context.store, NE, id, { collection: true })
+  const entity = await getWith<NE>(context.store, NE, id, { collection: true })
+  const collection = await getWith<CE>(context.store, CE, event.collectionId)
```

Comment:
You do not need getWith then

Reply:
are we sure about that? imo it won't save from inner join

Reply:
On line 18 for sure, you are not using collection from entity.
and on line 19 you are not getting anything with

Reply:
can confirm it need to get collection entity to be able to save entity, can't save collection with `getWith` 

**without**
<img width="588" alt="Capture d’écran 2023-11-10 à 9 37 18 AM" src="https://github.com/kodadot/stick/assets/9987732/57ed9682-b806-44af-870a-abf73057a08c">

**with**
<img width="588" alt="Capture d’écran 2023-11-10 à 9 37 12 AM" src="https://github.com/kodadot/stick/assets/9987732/72d0325b-9eca-4b60-a378-4f6e2cb6dc8e">

Reply:
should I replace `getWith` with `getOrFail`?